basiclti/tsugi-util/src/java/org/tsugi/HACK/HackMoodle.java

```diff
@@ -0,0 +1,72 @@
+
+package org.tsugi.HACK;
+
+import org.tsugi.basiclti.BasicLTIUtil;
+import org.json.simple.JSONValue;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONArray;
+
+public class HackMoodle {
+
+	/* In the IMS Dynamic Registration spec, messages_supported is a (kind of weird) array of objects:
+	 *
+	 * {
+	 *   "https://purl.imsglobal.org/spec/lti-platform-configuration": {
+     *      "product_family_code": "ExampleLMS",
+     *      "messages_supported": [
+     *        {"type": "LtiResourceLinkRequest"},
+     *        {"type": "LtiDeepLinkingRequest"}],
+     *      "variables": ["CourseSection.timeFrame.end", "CourseSection.timeFrame.begin", "Context.id.history", "ResourceLink.id.history"]
+	 *  }
+	 *
+	 * In Moodle (at least in 3.10) these come back as an array of strings.
+	 *
+	 * {
+	 *    "https://purl.imsglobal.org/spec/lti-platform-configuration": {
+     *      "product_family_code": "moodle",
+     *      "version": "3.10.9+ (Build: 20220129)",
+     *      "messages_supported": [
+     *          "LtiResourceLinkRequest",
+     *          "LtiDeepLinkingRequest"
+     *      ],
+     *    "variables": ["CourseSection.timeFrame.end", "CourseSection.timeFrame.begin", "Context.id.history", "ResourceLink.id.history"]
+	 *  }
+	 *
+	 *  Usage:
+	 *
+     *  body = org.tsugi.HACK.HackMoodle.hackOpenIdConfiguration(body);
+     *  openIDConfig = mapper.readValue(body, OpenIDProviderConfiguration.class);
+	 */
```

Comment:
Should this be removed?

Reply:
No - Sadly it is absolutely needed in the ProviderServlet so that SakaiPlus can interoperate with Moodle.  It is named with "hack" in the name to point out that it is a sad example of Moodle not following standards but we still need to be able to interoperate with them.

Reply:
I will fix the spaces while I am in here :)